src/solver/postprocess.jl

```diff
@@ -60,21 +54,38 @@ function _postprocess_root(x::SymbolicUtils.BasicSymbolic)
         return arguments(x)[1]
     end
 
-    # sqrt(M^2 * N) => M * sqrt(N)
+    # sqrt((N / D)^2 * M) => N / D * sqrt(M)
     if iscall(x) && (operation(x) === sqrt || operation(x) === ssqrt)
-        arg = arguments(x)[1]
-        if arg isa Integer
-            square, radical = big(1), big(1)
-            for (p, d) in collect(Primes.factor(abs(arg)))
+        function squarefree_decomp(x::Integer)
+            square, squarefree = big(1), big(1)
+            for (p, d) in collect(Primes.factor(abs(x)))
                 q, r = divrem(d, 2)
                 square *= p^q
-                radical *= p^r
+                squarefree *= p^r
             end
+            square, squarefree
+        end
+        arg = arguments(x)[1]
+        if arg isa Integer
+            square, squarefree = squarefree_decomp(arg)
             if arg < 0
                 square = im * square
             end
-            isone(radical) && return square
-            return square * Symbolics.term(Symbolics.operation(x), radical)
+            if !isone(square)
+                return square * Symbolics.term(Symbolics.operation(x), squarefree)
```

Comment:
Should we use `maketerm` to be more generic?

Reply:
We use `Symbolics.term` since `maketerm` simplifies arithmetic expressions that involve Base.MathConstants.pi and friends into floats

Reply:
I see.